packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -141,24 +179,31 @@ export async function launch(
     labels: METRIC_LABELS,
   });
 
-  // ////////////////////////////
-  // TODO: This is where we would add the scheduler.
-  //
-  // Note that the "bootstrap until no more progress" state will call this
-  // function without any arguments.
-  async function crankScheduler(maximumCranks) {
-    let now = Date.now();
+  async function crankScheduler(maximumCranks, clock = () => Date.now()) {
```

Comment:
Since this first `maximumCranks` argument is only used to choose the policy, and not by anything else, please do:

```suggestion
  async function crankScheduler(runBootstrap, clock = () => Date.now()) {
```

and then always supply `true` or `false` to `crankScheduler` in all its invocations.